server/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -81,6 +88,8 @@ public final class ShardRouting implements Writeable, ToXContentObject {
             : "replica shards always recover from primary";
         assert (currentNodeId == null) == (state == ShardRoutingState.UNASSIGNED)
             : "unassigned shard must not be assigned to a node " + this;
+        assert relocatingNodeId == null || (state != ShardRoutingState.STARTED && state != ShardRoutingState.UNASSIGNED)
```

Comment:
```suggestion
        assert relocatingNodeId == null || (state == ShardRoutingState.RELOCATING || state == ShardRoutingState.INITIALIZING)
```

Just wondering, in the above new comment you mentioned that state should be RELOCATING or INITIALIZING. Wouldn't it be better to assert these two states here as well? Rather than asserting the states it shouldn't be in.